### PR TITLE
Fixed compilation errors with giflib 5

### PR DESCRIPTION
**Pull request by Mailaender**
https://build.opensuse.org/package/view_file/Mono:Factory/libgdiplus0/libgdiplus0-giflib5.patch

### CODE REVIEW

File: src/gifcodec.c [hunk 1]
```diff
@@ -303,12 +310,19 @@ gdip_load_gif_image (void *stream, GpImage **image, BOOL from_file)
...
 		gif = DGifOpen (stream, &gdip_gif_inputfunc);
 	}
-	
+#else
+	if (from_file)
+		gif = DGifOpen(stream, &gdip_gif_fileinputfunc, NULL);
+	else
+		gif = DGifOpen(stream, &gdip_gif_inputfunc, NULL);
+#endif
+
```
Comment (Mailaender) [inline src/gifcodec.c]: I don't really know C nor what macros are. Filed https://github.com/mono/libgdiplus/pull/31 instead.

**Comment by Mailaender** [comment]
I don't really know enough C to make this more readable. I believe the original source was https://projects.archlinux.org/svntogit/packages.git/plain/trunk/libgdiplus0-giflib5.patch?h=packages/libgdiplus

**Comment by Mailaender** [comment]
Rebased and broke the compilation for @openSUSE 13.2 :anger:

File: src/gifcodec.c (hunk 1 above)
Comment (akoeplinger) [inline src/gifcodec.c]: same as what you did on line 49 and 50. But that's just my personal opinion here :)

**Comment by Mailaender** [comment]
Can't get this to work with macros. Superseded by https://github.com/mono/libgdiplus/pull/32.

**Pull request closed without merging**